python/ray/ml/preprocessor.py

```diff
@@ -60,6 +64,8 @@ def fit_transform(self, dataset: Dataset) -> Dataset:
         Returns:
             ray.data.Dataset: The transformed Dataset.
         """
+        assert self._is_fittable, "One is expected to call `should_fit` before `fit`."
```

Comment:
this error message looks weird. why don't you check:
assert self.should_fit() here as well?